Tensor mask changes

SundarRajan28 opened the pull request:
Added changes for supporting tensor mask trainings

fiona-gladwin committed:
ROI pipeline changes

Add support to allocate ROI in ring buffer
Add API to read and write into ROI buffers
Introduce API to copy and swap the ROI

fiona-gladwin committed:
Remove rocALTensorInfo constructor and destructor

Change ROI to a shared pointer

fiona-gladwin committed:
ROI fix

Remove separate API in ring buffer to get ROI buffers
Return the output and ROI buffers together as pair
Replace swap with reset for ROI shared pointer

SundarRajan28 committed:
Fixing warp affine test case in python unit tests

SundarRajan28 committed:
Merge remote-tracking branch 'origin/lk/rocalTensor_pybind' into PR_tensor_mask

SundarRajan28 committed:
Merge remote-tracking branch 'origin/PR_metanode_reader_changes' into PR_tensor_mask

SundarRajan28 committed:
Adding changes to RMN augmentation

fiona-gladwin committed:
Change ROCAL_NHWC to ROCAL_NONE in default arguments

fiona-gladwin committed:
Change argument name

SundarRajan28 committed:
Uncommenting metagraph nodes for augmentations

SundarRajan28 committed:
Uncommenting metagraph nodes for augmentations

SundarRajan28 committed:
Uncommenting metagraph nodes for augmentations

fiona-gladwin committed:
Add fix for grayscale outputs

fiona-gladwin committed:
Merge branch 'PR_tensor_node_augmentations' into lk/rocalTensor_PR

fiona-gladwin committed:
change copy right year in tensor augmentations

fiona-gladwin committed:
Fix build issues

fiona-gladwin committed:
Fix build issues

fiona-gladwin committed:
Merge branch 'PR_vx_tensor_augmentations' into lk/rocalTensor_PR

fiona-gladwin committed:
Merge branch 'PR_tensor_ROI' into lk/rocalTensor_PR

SundarRajan28 committed:
Moving ROI updation to RMN meta node

fiona-gladwin committed:
change unsigned to int in for loop

SundarRajan28 committed:
Merge remote-tracking branch 'origin/lk/rocalTensor_PR' into PR_metanode_reader_changes

SundarRajan28 committed:
Merge remote-tracking branch 'origin/PR_metanode_reader_changes' into PR_python_examples_notebooks

SundarRajan28 committed:
Merge remote-tracking branch 'origin/PR_metanode_reader_changes' into PR_tensor_mask

SundarRajan28 committed:
Merge branch 'PR_tensor_mask' of https://github.com/fiona-gladwin/MIVisionX into PR_tensor_mask

SundarRajan28 committed:
Merge remote-tracking branch 'origin/lk/rocalTensor_PR' into lk/rocalTensor_pybind

SundarRajan28 committed:
Merge remote-tracking branch 'origin/lk/rocalTensor_pybind' into PR_metanode_reader_changes

fiona-gladwin committed:
code cleanup

SundarRajan28 committed:
Renaming pipeline functions

SundarRajan28 committed:
Renaming pipeline functions

fiona-gladwin committed:
Modify unit test

Add support to check all augmentations with file source reader
Enable remianing augmentations in unit test

fiona-gladwin committed:
Add parameters description

fiona-gladwin committed:
Changed params description  wrt Doxygen

SundarRajan28 committed:
Resolving review comments

Commit:
Minor code change in python examples

fiona-gladwin committed:
Merge branch 'PR_tensor_mask' of https://github.com/fiona-gladwin/MIVisionX into PR_tensor_mask

fiona-gladwin committed:
Merge branch 'PR_vx_tensor_augmentations' into lk/rocalTensor_PR

fiona-gladwin committed:
Merge branch 'PR_tensor_node_augmentations' into lk/rocalTensor_PR

fiona-gladwin committed:
Fix warnings

fiona-gladwin committed:
Minor changes

fiona-gladwin committed:
Fix warnings

fiona-gladwin committed:
Merge branch 'PR_tensor_ROI' into lk/rocalTensor_PR

SundarRajan28 committed:
Merge remote-tracking branch 'origin/lk/rocalTensor_PR' into lk/rocalTensor_pybind

SundarRajan28 committed:
Merge remote-tracking branch 'origin/lk/rocalTensor_PR' into PR_metanode_reader_changes

SundarRajan28 committed:
Merge remote-tracking branch 'origin/lk/rocalTensor_PR' into PR_python_examples_notebooks

SundarRajan28 committed:
Merge remote-tracking branch 'origin/lk/rocalTensor_PR' into PR_tensor_mask

SundarRajan28 committed:
Merge branch 'PR_tensor_mask' of https://github.com/fiona-gladwin/MIVisionX into PR_tensor_mask

SundarRajan28 committed:
Reverting comparison file changes in C++ unit tests folder

fiona-gladwin committed:
Fix build issue wrt merge

fiona-gladwin committed:
Add changes to parse image ID

Add map to relate image names with ID
Add changes to API to get the image ID direcly

SundarRajan28 committed:
Adding changes to copy_data wrappers

SundarRajan28 committed:
Adding changes to copy_data wrappers

SundarRajan28 committed:
Adding changes to copy_data wrappers

SundarRajan28 committed:
Added class remapping variable for coco meta reader

SundarRajan28 committed:
Resolving review comments

SundarRajan28 committed:
Removing extra lines in video pipeline example

SundarRajan28 committed:
Adding class_remapping function to metadata config

SundarRajan28 committed:
Removing extra lines and whitespaces

SundarRajan28 committed:
Formatting changes

fiona-gladwin committed:
Merge branch 'lk/rocalTensor' of https://github.com/LakshmiKumar23/MIVisionX into PR_metadata_fix

fiona-gladwin committed:
Merge branch 'lk/rocalTensor' of https://github.com/LakshmiKumar23/MIVisionX into PR_vx_tensor_augmentations

fiona-gladwin committed:
Merge branch 'lk/rocalTensor' of https://github.com/LakshmiKumar23/MIVisionX into PR_tensor_node_augmentations

SundarRajan28 committed:
Resolving review comments and formatting changes

SundarRajan28 committed:
Reverting file changes for iterators and video example

SundarRajan28 committed:
Formatting and renaming changes from tensor_pybind_PR

SundarRajan28 committed:
Adding aspect ratio grouping option for COCO reader

SundarRajan28 committed:
Merge remote-tracking branch 'origin/lk/rocalTensor_PR' into lk/rocalTensor_pybind

SundarRajan28 committed:
Merge remote-tracking branch 'origin/tensor_pybind_PR' into lk/rocalTensor_pybind

SundarRajan28 committed:
Merge remote-tracking branch 'origin/lk/rocalTensor_pybind' into PR_metanode_reader_changes

SundarRajan28 committed:
Reverting files to base branch

SundarRajan28 committed:
Adding config changes for python unittests

SundarRajan28 committed:
Adding config changes for python unittests

SundarRajan28 committed:
Adding missing comma in pybind

SundarRajan28 committed:
Removing ROI functions from augmentation nodes and adding it in node.h

SundarRajan28 committed:
Resolving review comments and formatting changes

SundarRajan28 committed:
Fixing warp affine test case

SundarRajan28 committed:
Removing failing test cases from python unit tests

fiona-gladwin committed:
Add comments to clone API

fiona-gladwin committed:
Modified all tensor API descriptions

fiona-gladwin committed:
Minor fix

fiona-gladwin committed:
Modify API description

Mark all params as input
Add description for each augmentation
Add return type in description

fiona-gladwin committed:
Merge branch 'PR_metadata_fix' into lk/rocalTensor_PR

fiona-gladwin committed:
Merge branch 'PR_vx_tensor_augmentations' into lk/rocalTensor_PR

fiona-gladwin committed:
Merge branch 'PR_tensor_node_augmentations' into lk/rocalTensor_PR

fiona-gladwin committed:
Minor fix

fiona-gladwin committed:
Minor changes

fiona-gladwin committed:
Merge branch 'PR_tensor_node_augmentations' into lk/rocalTensor_PR

SundarRajan28 committed:
Merge remote-tracking branch 'origin/lk/rocalTensor_PR' into PR_metanode_reader_changes

SundarRajan28 committed:
Merge remote-tracking branch 'origin/lk/rocalTensor_PR' into lk/rocalTensor_pybind

SundarRajan28 committed:
Merge remote-tracking branch 'origin/lk/rocalTensor_pybind' into PR_metanode_reader_changes

SundarRajan28 committed:
Reverting ROI changes PR

SundarRajan28 committed:
Reverting ROI changes PR

SundarRajan28 committed:
Merge remote-tracking branch 'origin/lk/rocalTensor_pybind' into PR_metanode_reader_changes

swetha097 commented on rocAL/rocAL/include/api/rocal_api_augmentation.h:
```diff
@@ -24,76 +24,100 @@ THE SOFTWARE.
+ * Frames can be repeated or dropped in the new_order.
+ * \ingroup group_rocal_augmentations
+ * \note: Accepts U8 and RGB24 input.
+ * \param context context for the pipeline.
```
Add the correct arg names here p_context & p_input
Change the explanations to tell that its a pointer variable

swetha097 commented on rocAL/rocAL/include/api/rocal_api_augmentation.h:
```diff
@@ -24,76 +24,100 @@ THE SOFTWARE.
+ * \return
+ */
+extern "C" RocalTensor ROCAL_API_CALL rocalSequenceRearrange(RocalContext p_context, RocalTensor p_input,
                                                               std::vector<unsigned int>& new_order,
```
Note the Aligment of the arguments

swetha097 commented on rocAL/rocAL/include/api/rocal_api_augmentation.h:
```diff
@@ -24,76 +24,100 @@ THE SOFTWARE.
-                                                   RocalTensorOutputType output_datatype = ROCAL_UINT8);
+/*! \brief Accepts U8 and RGB24 input.
+ * \ingroup group_rocal_augmentations
+ * \param context context for the pipeline.
```
Make it either p_context / context - Keep it uniform across the file

swetha097 commented on rocAL/rocAL/include/api/rocal_api_augmentation.h:
```diff
@@ -24,76 +24,100 @@ THE SOFTWARE.
+ * \param max_size Limits the size of the resized tensor.
+ * \param resize_shorter The length of the shorter dimension of the image.
+ * \param resize_longer The length of the larger dimension of the image.
+ * \param interpolation_type The type of interpolation to be used for resize.
```
mirror arg exaplantion is missing

swetha097 commented on rocAL/rocAL/include/api/rocal_api_augmentation.h:
```diff
@@ -104,16 +128,22 @@ extern "C" RocalTensor ROCAL_API_CALL rocalCropResize(RocalContext context, Roca
+ * \param input Input Rocal tensor
+ * \param dest_width output width
+ * \param dest_height output height
+ * \param is_output True: the output tensor is needed by user and will be copied to output buffers using the data
```
is_output True : Remove this True

swetha097 reviewed (commented):
In rocal_api_augmentation.py file - Some of the args exaplanation start with a CAPs letter & some small. Please Keep it CAPs . Maintianin this uniformly in file.

swetha097 reviewed (commented):
Some of the args exaplanation start with a CAPs letter & some small. Please Keep it CAPs . Maintianin this uniformly in file.

swetha097 commented on rocAL/rocAL/include/api/rocal_api_augmentation.h:
```diff
@@ -24,76 +24,100 @@ THE SOFTWARE.
+ * \param is_output True: the output tensor is needed by user and will be copied to output buffers using the data
+ * transfer API calls. False: the output tensor is just an intermediate tensor, user is not interested in
+ * using it directly. This option allows certain optimizations to be achieved.
+ * \param area Target area for the crop
```
aspect_ratio explanation missing

swetha097 commented on rocAL/rocAL/include/api/rocal_api_augmentation.h:
```diff
@@ -122,294 +152,384 @@ extern "C" RocalTensor ROCAL_API_CALL rocalCropResizeFixed(RocalContext context,
+ * \param dest_width output width
+ * \param dest_height output height
+ * \param is_output Is the output tensor part of the graph output
+ * \param angle The rotation angle value in degrees.
```
Change the order of the params explanation

swetha097 commented on rocAL/rocAL/include/api/rocal_api_augmentation.h:
```diff
@@ -122,294 +152,384 @@ extern "C" RocalTensor ROCAL_API_CALL rocalCropResizeFixed(RocalContext context,
-/// \param is_output
-/// \return
+/*! \brief Accepts U8 and RGB24 input.
+ * \ingroup group_rocal_augmentations
```
Check the order of parameter explanation

swetha097 reviewed (commented):
In rocal_api_augmentation.cpp
Check the order of the Params Explanation added. Many seem jumbled up.

swetha097 commented on rocAL/rocAL/include/api/rocal_api_meta_data.h:
```diff
@@ -150,6 +150,7 @@ extern "C" RocalTensorList ROCAL_API_CALL rocalGetBoundingBoxLabel(RocalContext
 extern "C" RocalTensorList ROCAL_API_CALL rocalGetBoundingBoxCords(RocalContext rocal_context);
 
 extern "C" void ROCAL_API_CALL rocalGetImageSizes(RocalContext rocal_context, int* buf);
+extern "C" void ROCAL_API_CALL rocalGetROIImageSizes(RocalContext rocal_context, int* buf);
```
Add an explanation for the arguments

swetha097 commented on utilities/rocAL/rocAL_unittests/rocAL_unittests.cpp:
```diff
@@ -771,7 +771,17 @@ int test(int test_case, int reader_type, const char *path, const char *outName,
+        std::vector<float> mean = {128, 128, 128};
+        std::vector<float> std_dev = {1.2, 1.2, 1.2};
+        std::cout << ">>>>>>> Running "
+                  << " Resize Mirror Normalize " << std::endl;
```
Are we not having a test case for MIN_MAX scaling mode?

swetha097 commented on rocAL/rocAL/source/meta_data/meta_node_resize_mirror_normalize.cpp:
```diff
@@ -90,6 +90,11 @@ void ResizeMirrorNormalizeMetaNode::update_parameters(pMetaDataBatch input_meta_
         }
+        // get roi width and height of output image
+        auto img_roi_size = input_meta_data->get_img_roi_sizes_batch()[i];
+        img_roi_size.w = output_roi[i].x2;
+        img_roi_size.h = output_roi[i].y2;
```
Are we storing the dst ROIs in x2 & y2?
Even for dst we would have x1,y1 and x2,y2 right?

swetha097 commented on rocAL/rocAL/source/readers/image/coco_file_source_reader.cpp:
```diff
@@ -86,9 +86,55 @@ Reader::Status COCOFileSourceReader::initialize(ReaderConfig desc)
-    //shuffle dataset if set
-    if (ret == Reader::Status::OK && _shuffle)
-        std::random_shuffle(_file_names.begin(), _file_names.end());
+
```
I am seeing this same code is repeated twice in "reset()"  & "initialize".
Put this part of the code in a function block & call it when required

swetha097 commented on rocAL/rocAL/source/readers/image/coco_file_source_reader.cpp:
```diff
@@ -86,9 +86,55 @@ Reader::Status COCOFileSourceReader::initialize(ReaderConfig desc)
-        std::random_shuffle(_file_names.begin(), _file_names.end());
+
+    if (_meta_data_reader && _meta_data_reader->aspect_ratio_grouping())
+    {
```
Please correct the bracketing style according to goggle standards

swetha097 commented on rocAL/rocAL/source/readers/image/coco_file_source_reader.cpp:
```diff
@@ -86,9 +86,55 @@ Reader::Status COCOFileSourceReader::initialize(ReaderConfig desc)
+                  { return lop.second < rop.second; });
+
+        // extract sorted file_names
+        std::vector<std::string> sorted;
```
Rename this to something more readable like sorted_file_names instead of just sorted

swetha097 commented on utilities/rocAL/rocAL_unittests/testAllScripts.sh:
```diff
@@ -79,89 +79,84 @@ do
     do 
-        # FileSource Reader - Tensor
+        # FileSource Reader
+        ./rocAL_unittests 0 "$image_path" "${output_path}LensCorrection_${rgb_name[$rgb]}_${device_name}" $width $height 45 $device $rgb 0 $display
```
Are these test cases supposed to be a part of this PR ?

swetha097 reviewed (commented):
Added the first round of review, Please address them

SundarRajan28 committed:
Resolving review comments

SundarRajan28 committed:
Fixing spacing issues in CMN node

SundarRajan28 committed:
Adding metanode for flip augmentation

SundarRajan28 committed:
Removing torch dependency and codacy warnings

SundarRajan28 committed:
Removing torch dependency and codacy warnings

SundarRajan28 committed:
Merge branch 'PR_metanode_reader_changes' into PR_python_examples_notebooks

SundarRajan28 committed:
Removing unused variables in tf iterator

SundarRajan28 committed:
Removing unused variables in tf iterator

SundarRajan28 committed:
Merge branch 'PR_metanode_reader_changes' into PR_python_examples_notebooks

SundarRajan28 committed:
Minor code changes to jupyter notebooks

fiona-gladwin commented:
@SundarRajan28 Please resolve merge conflicts.
Sync your base branch and PR_tensor_mask branch with changes from lk/rocalTensor_PR so that will reduce all the vx_rpp files
Please update the PR

fiona-gladwin commented on rocAL/rocAL/include/meta_data/meta_data.h:
```diff
@@ -211,18 +214,22 @@ class MetaDataInfoBatch {
     }
     void insert(MetaDataInfoBatch &other) {
         img_sizes.insert(img_sizes.end(), other.img_sizes.begin(), other.img_sizes.end());
+        img_roi_sizes.insert(img_roi_sizes.end(), other.img_roi_sizes.begin(), other.img_roi_sizes.end());
```
Add at the end

fiona-gladwin commented on rocAL/rocAL/include/meta_data/meta_data_reader.h:
```diff
@@ -57,28 +57,36 @@ struct MetaDataConfig
     unsigned out_img_height() const { return _out_img_height; }
     void set_out_img_width(unsigned out_img_width) { _out_img_width = out_img_width; }
     void set_out_img_height(unsigned out_img_height) { _out_img_height = out_img_height; }
+    void set_class_remapping(bool avoid_class_remapping) { _avoid_class_remapping = avoid_class_remapping; }
```
The name of API suggests we are setting the class remapping
The name of the variable suggests we are avoiding class remapping

set class remapping would mean we are enabling the remapping right? Not avoiding it. Check how this can be made meaningful

SundarRajan28 replied:
Changed it to set_avoid_class_remapping

fiona-gladwin commented on rocAL/rocAL/include/meta_data/meta_data_reader.h:
```diff
@@ -88,9 +96,12 @@ class MetaDataReader
     virtual void init(const MetaDataConfig& cfg, pMetaDataBatch meta_data_batch) = 0;
     virtual void read_all(const std::string& path) = 0;// Reads all the meta data information
     virtual void lookup(const std::vector<std::string>& image_names) = 0;// finds meta_data info associated with given names and fills the output
+    virtual ImgSize lookup_image_size(const std::string& image_name) { return {}; };
```
Semicolon at end not required

fiona-gladwin commented on rocAL/rocAL/include/pipeline/master_graph.h:
```diff
@@ -101,7 +101,7 @@ class MasterGraph
     std::vector<rocalTensorList *> create_label_reader(const char *source_path, MetaDataReaderType reader_type);
     std::vector<rocalTensorList *> create_video_label_reader(const char *source_path, MetaDataReaderType reader_type, unsigned sequence_length, unsigned frame_step, unsigned frame_stride, bool file_list_frame_num = true);
-    std::vector<rocalTensorList *> create_coco_meta_data_reader(const char *source_path, bool is_output, MetaDataReaderType reader_type, MetaDataType label_type, bool ltrb_bbox = true, bool is_box_encoder = false, float sigma = 0.0, unsigned pose_output_width = 0, unsigned pose_output_height = 0);
+    std::vector<rocalTensorList *> create_coco_meta_data_reader(const char *source_path, bool is_output, MetaDataReaderType reader_type, MetaDataType label_type, bool ltrb_bbox = true, bool is_box_encoder = false, bool avoid_class_remapping=false, bool aspect_ratio_grouping = false, float sigma = 0.0, unsigned pose_output_width = 0, unsigned pose_output_height = 0);
```
Space before and after = operator
This API is too long can we add in multiple lines?

fiona-gladwin commented on rocAL/rocAL/source/readers/image/coco_file_source_reader.cpp:
```diff
@@ -86,9 +86,55 @@ Reader::Status COCOFileSourceReader::initialize(ReaderConfig desc)
+        // zip the two vectors
+        std::vector<std::pair<std::string, float>> zipped(_file_names.size());
+        for (size_t i = 0; i < _file_names.size(); i++)
+            zipped[i] = std::make_pair(_file_names[i], _aspect_ratios[i]);
```
Can we not do this as part of previous for loop?

fiona-gladwin commented on rocAL/rocAL/source/readers/image/coco_file_source_reader.cpp:
```diff
@@ -86,9 +86,55 @@ Reader::Status COCOFileSourceReader::initialize(ReaderConfig desc)
+            zipped[i] = std::make_pair(_file_names[i], _aspect_ratios[i]);
+
+        // sort according to aspect ratios
+        std::sort(zipped.begin(), zipped.end(), [](auto &lop, auto &rop)
```
What is lop and rop?

fiona-gladwin commented on rocAL/rocAL/source/readers/image/coco_file_source_reader.cpp:
```diff
@@ -86,9 +86,55 @@ Reader::Status COCOFileSourceReader::initialize(ReaderConfig desc)
+
+        _file_names = sorted;
+        std::sort(_aspect_ratios.begin(), _aspect_ratios.end());
+        auto mid = std::upper_bound(_aspect_ratios.begin(), _aspect_ratios.end(), 1.0f) - _aspect_ratios.begin();
```
Explain this line in comment

fiona-gladwin commented on rocAL/rocAL/source/readers/image/coco_file_source_reader.cpp:
```diff
@@ -86,9 +86,55 @@ Reader::Status COCOFileSourceReader::initialize(ReaderConfig desc)
+        // shuffle dataset if set
+        if (ret == Reader::Status::OK && _shuffle)
+        {
+            std::random_shuffle(_file_names.begin(), _file_names.begin() + mid);
```
Why is it done twice?

fiona-gladwin commented on rocAL/rocAL/source/readers/image/coco_file_source_reader.cpp:
```diff
@@ -86,9 +86,55 @@ Reader::Status COCOFileSourceReader::initialize(ReaderConfig desc)
+        auto mid = std::upper_bound(_aspect_ratios.begin(), _aspect_ratios.end(), 1.0f) - _aspect_ratios.begin();
+
+        // shuffle dataset if set
+        if (ret == Reader::Status::OK && _shuffle)
```
Explain this portion for better understanding

fiona-gladwin commented on rocAL/rocAL/source/readers/image/coco_file_source_reader.cpp:
```diff
@@ -184,7 +230,50 @@ int COCOFileSourceReader::release()
-    if (_shuffle)
+    _aspect_ratios.clear();
+
+    if (_meta_data_reader && _meta_data_reader->aspect_ratio_grouping())
```
Why should we do it every time we reset can you please explain?

fiona-gladwin commented on rocAL/rocAL/source/readers/image/coco_file_source_reader.cpp:
```diff
@@ -184,7 +230,50 @@ int COCOFileSourceReader::release()
+            _file_names = shuffled_filenames;
+        }
+    }
+    else 
```
Check if shuffle is enabled and then shuffle
Enclose within a code block.

fiona-gladwin commented on rocAL/rocAL/source/meta_data/meta_node_resize_mirror_normalize.cpp:
```diff
@@ -90,6 +90,11 @@ void ResizeMirrorNormalizeMetaNode::update_parameters(pMetaDataBatch input_meta_
             bb_labels.push_back(labels_buf[j]);
         }
+        // get roi width and height of output image
+        auto img_roi_size = input_meta_data->get_img_roi_sizes_batch()[i];
```
Will this be updated in other nodes? or not required?

SundarRajan28 replied:
Not required unless any other training requires ROI info for the final augmentation

fiona-gladwin commented on rocAL/rocAL/source/meta_data/coco_meta_data_reader.cpp:
```diff
@@ -369,10 +381,14 @@ void COCOMetaDataReader::read_all(const std::string &path)
         {
             auto _it_label = _label_info.find(bb_labels[i]);
-            int cnt_idx = _it_label->second;
+            if (_avoid_class_remapping)
```
When you have if else, we need to use brackets check google coding guidelines

fiona-gladwin replied:
Can we use a ternary operator here?

fiona-gladwin commented on rocAL/rocAL/source/meta_data/coco_meta_data_reader.cpp:
```diff
@@ -32,6 +32,8 @@ using namespace std;
 void COCOMetaDataReader::init(const MetaDataConfig &cfg, pMetaDataBatch meta_data_batch)
 {
     _path = cfg.path();
+    _avoid_class_remapping = cfg.class_remapping();
```
Class remapping we need not set to this pointer?
Have you tested and validated these?

SundarRajan28 replied:
_avoid_class_remapping is a variable in the metadata reader whereas the aspect ratio grouping is set in the metadata reader using this function

swetha097 replied:
Why is this difference followed for the 2 variables ?

SundarRajan28 replied:
avoid_class_remapping is used within this class itself so its accessible but we need aspect_ratio_grouping variable accessible in coco_file_source_reader. Private variables won't be accessible outside class so added a public function to get the value and use it

fiona-gladwin commented on rocAL/rocAL/source/meta_data/coco_meta_data_reader.cpp:
```diff
@@ -183,6 +193,7 @@ void COCOMetaDataReader::read_all(const std::string &path)
     {
         if (0 == std::strcmp(key, "images"))
         {
+            int original_id;
```
VCan we name to image_id

fiona-gladwin commented on rocAL/rocAL/source/api/rocal_api_augmentation.cpp:
```diff
@@ -658,7 +660,10 @@ ROCAL_API_CALL rocalResizeMirrorNormalize(
             }
         }
-
+        if(scaling_mode == RocalResizeScalingMode::ROCAL_SCALING_MODE_MIN_MAX) {
```
Add comment why this is added

fiona-gladwin commented on rocAL/rocAL/source/api/rocal_api_augmentation.cpp:
```diff
@@ -675,8 +680,9 @@ ROCAL_API_CALL rocalResizeMirrorNormalize(
     } catch(const std::exception& e) {
         context->capture_error(e.what());
-        ERR(e.what())
+        ERR(e.what());
```
Remove semicolon here

SundarRajan28 committed:
Merge branch 'PR_metanode_reader_changes' into PR_tensor_mask

SundarRajan28 committed:
Fixing bugs in metanodes w.r.t rescaling

SundarRajan28 committed:
Resolving review comments and formatting changes

SundarRajan28 committed:
Merge branch 'tensor_pybind_PR' into lk/rocalTensor_pybind

SundarRajan28 committed:
Merge branch 'lk/rocalTensor_pybind' into PR_metanode_reader_changes

SundarRajan28 committed:
Resolving review comments

SundarRajan28 committed:
Formatting changes to readers.py

SundarRajan28 committed:
Adding timingInfo print statements back to tf iterator

SundarRajan28 committed:
Merge branch 'PR_metanode_reader_changes' into PR_python_examples_notebooks

SundarRajan28 committed:
Updated prefetch_queue_depth example

SundarRajan28 committed:
Fixing wrong function name in pipeline.py

SundarRajan28 committed:
Fixing wrong function name in pipeline.py

SundarRajan28 committed:
Merge branch 'PR_metanode_reader_changes' into PR_python_examples_notebooks

SundarRajan28 committed:
Updated caffe2 reader python example

SundarRajan28 committed:
Updated caffe reader python example

SundarRajan28 committed:
Updating pytorch classification reader

SundarRajan28 committed:
Updating TF python examples

SundarRajan28 committed:
Updating video pipeline python example

SundarRajan28 committed:
Adding API changes to rocal_api_pipeline python example

fiona-gladwin committed:
Fixing context release issue

fiona-gladwin committed:
Merge branch 'PR_tensor_unittest1' into PR_tensor_node_augmentations

fiona-gladwin committed:
Merge branch 'lk/rocalTensor' of https://github.com/LakshmiKumar23/MIVisionX into PR_tensor_node_augmentations

fiona-gladwin committed:
Merge branch 'lk/rocalTensor' of https://github.com/LakshmiKumar23/MIVisionX into PR_tensor_node_augmentations

fiona-gladwin committed:
Minor change

fiona-gladwin committed:
Add single API to modify width and height based on dims

swetha097 committed:
Port rocal_basic_test.cpp

swetha097 committed:
Port Cifar10 dataloader

swetha097 committed:
Port tf dataloader

swetha097 committed:
Fix labels in tf porting

swetha097 committed:
Port the performance tests

swetha097 committed:
Add Video Unit Test and data_loader_mtc.pp

swetha097 committed:
Port the image augmentation app

swetha097 committed:
Clean up code in Image_augmentation app

kamalasubha commented on rocAL/rocAL/source/readers/image/coco_file_source_reader.cpp:
```diff
@@ -184,8 +227,48 @@ int COCOFileSourceReader::release()
+        };
+
+        // zip the two vectors
+        std::vector<std::pair<std::string, float>> zipped(_file_names.size());
```
Rename zipped with some other param

kamalasubha commented on rocAL/rocAL/source/api/rocal_api_meta_data.cpp:
```diff
@@ -387,6 +386,37 @@ ROCAL_API_CALL rocalGetImageSizes(RocalContext p_context, int* buf)
+{
+    if (!p_context)
+    {
+        ERR("Invalid rocal context passed to rocalGetROIImageSizes")
```
can we use throw here?

kamalasubha commented on rocAL/rocAL/source/meta_data/coco_meta_data_reader.cpp:
```diff
@@ -318,12 +333,9 @@ void COCOMetaDataReader::read_all(const std::string &path)
-                std::string file_name = str + ".jpg";
 
-                auto it = _map_img_sizes.find(file_name);
+                auto itr = _map_img_names.find(id);
```
Rename itr and it variable with better naming

kamalasubha commented on rocAL/rocAL/source/readers/image/coco_file_source_reader.cpp:
```diff
@@ -86,9 +86,52 @@ Reader::Status COCOFileSourceReader::initialize(ReaderConfig desc)
-    if (ret == Reader::Status::OK && _shuffle)
-        std::random_shuffle(_file_names.begin(), _file_names.end());
+
+    if (_meta_data_reader && _meta_data_reader->aspect_ratio_grouping()) {
```
Please add comments to the function

kamalasubha commented on rocAL/rocAL/source/readers/image/coco_file_source_reader.cpp:
```diff
@@ -184,8 +227,48 @@ int COCOFileSourceReader::release()
+        std::vector<std::string> sorted;
+        std::transform(zipped.begin(), zipped.end(), std::back_inserter(sorted), [](auto &pair)
+                        { return pair.first; });
+
```
since most of the operations seems same, can we create one function as common and call from initialize and reset?

SundarRajan28 replied:
Optimized the AR grouping and removed the repeated operations

SundarRajan28 committed:
Adding tfpets training example

fiona-gladwin committed:
Minor change

fiona-gladwin committed:
Add API in tensor info to modify strides

swetha097 committed:
Changes to accomodate Video Reader

SundarRajan28 committed:
Resolving review comments

swetha097 committed:
Minor changes in image_augmentation app

swetha097 committed:
Revert c17cad844c4052ada39636950915558c3721bf02

swetha097 committed:
Chnage imwrite to imshow

swetha097 committed:
Some minor changes in rocal_basic_test

swetha097 committed:
Change backt the processing device

swetha097 committed:
Minor changes

swetha097 committed:
Minor changes in rocAL_dataloader.cpp

SundarRajan28 committed:
Fixing bug with metadata updation for cascaded geometric augmentations

SundarRajan28 committed:
Fixing bug with metadata updation for cascaded geometric augmentations

SundarRajan28 committed:
Fixing bug with metadata updation for cascaded geometric augmentations

SundarRajan28 committed:
Adding missing type for scaling mode

SundarRajan28 committed:
Fixing bb_labels wrong function name

SundarRajan28 committed:
Fixing bug with image ID updation for coco mask pipeline

fiona-gladwin committed:
Add changes to color format API

SundarRajan28 committed:
Adding copyOutTensor to mask pipeline

fiona-gladwin committed:
Introduce new tensorInfo structure to combine args in dataloader

fiona-gladwin committed:
Merge branch 'lk/rocalTensor' of https://github.com/LakshmiKumar23/MIVisionX into PR_tensor_node_augmentations

swetha097 commented on rocAL/rocAL/include/api/rocal_api_meta_data.h:
```diff
@@ -60,8 +60,13 @@ extern "C" RocalMetaData ROCAL_API_CALL rocalCreateTFReaderDetection(RocalContex
 ///
 /// \param rocal_context
 /// \param source_path path to the coco json file
+/// \param mask enable polygon masks
```
is_output desciption is missing

SundarRajan28 replied:
Currently is_output param for all reader functions are unused so its not been added as param anywhere. I'll check with Shobana and confirm once before adding the param desc

swetha097 commented on rocAL/rocAL/include/api/rocal_api_meta_data.h:
```diff
@@ -60,8 +60,13 @@ extern "C" RocalMetaData ROCAL_API_CALL rocalCreateTFReaderDetection(RocalContex
 /// \param rocal_context
 /// \param source_path path to the coco json file
+/// \param mask enable polygon masks
+/// \param ltrb If set to True, bboxes are returned as [left, top, right, bottom]. If set to False, the bboxes are returned as [x, y, width, height
```
Close with ]

swetha097 commented on rocAL/rocAL_pybind/amd/rocal/pipeline.py:
```diff
@@ -173,6 +173,10 @@ def define_graph(self):
     def get_handle(self):
         return self._handle
 
+    def copyToExternalTensor(self, array,  multiplier, offset, reverse_channels, tensor_format, tensor_dtype):
```
Are we introducing copyToExternalTensor in this mask PR?

Please align the arguments

SundarRajan28 replied:
copyToExternalTensor C++ implementation is already introduced but we didn't need the pybind since copy_data was enough for classification trainings. Mask GPU trainings were lagging in perf so Shobana asked to introduce copyTensor to remove the perf difference

swetha097 commented on rocAL/rocAL_pybind/amd/rocal/pipeline.py:
```diff
@@ -231,12 +235,18 @@ def get_image_id(self, array):
     def get_bounding_box_cords(self):
         return b.getBoundingBoxCords(self._handle)
 
+    def get_mask_count(self, array):
```
Add all the new functions at the end at a single place

SundarRajan28 replied:
I've added the metadata functions together and the info functions (img_sizes()) together to make it easier to group functions

swetha097 commented on rocAL/rocAL_pybind/rocal_pybind.cpp:
```diff
@@ -91,6 +91,20 @@ namespace rocal{
     }
 
+    py::object wrapper_tensor(RocalContext context, py::object p,
+                                RocalTensorLayout tensor_format, RocalTensorOutputType tensor_output_type, float multiplier0,
```
Align the parameters

swetha097 commented on rocAL/rocAL_pybind/rocal_pybind.cpp:
```diff
@@ -91,6 +91,20 @@ namespace rocal{
+    {
+        auto ptr = ctypes_void_ptr(p);
+        // call pure C++ function
+        int status = rocalToTensor(context, ptr, tensor_format, tensor_output_type, multiplier0,
```
Align the parameters

swetha097 commented on rocAL/rocAL_pybind/rocal_pybind.cpp:
```diff
@@ -377,6 +392,11 @@ namespace rocal{
+        m.def("getROIImgSizes", [](RocalContext context, py::array_t<int> array) {
+            auto buf = array.request();
+            int* ptr = static_cast<int *>(buf.ptr);
+            rocalGetROIImageSizes(context,ptr);
```
Space after ","

swetha097 commented on rocAL/rocAL_pybind/rocal_pybind.cpp:
```diff
@@ -395,6 +415,7 @@ namespace rocal{
         m.def("getIntValue", &rocalGetIntValue);
         m.def("getFloatValue", &rocalGetFloatValue);
         // rocal_api_data_transfer.h
+        m.def("rocalToTensor", &wrapper_tensor);
```
Please give a suitable name here - like wrapper_to_tensor

swetha097 commented on rocAL/rocAL_pybind/rocal_pybind.cpp:
```diff
@@ -448,6 +469,44 @@ namespace rocal{
+            auto mask_count_buf = mask_count.request();
+            int* mask_count_ptr = static_cast<int *>(mask_count_buf.ptr);
+            for(int i = 0; i < bbox_labels->size(); i++) // nbatchSize
+            {
```
Change the bracketing style

swetha097 commented on rocAL/rocAL_pybind/rocal_pybind.cpp:
```diff
@@ -448,6 +469,44 @@ namespace rocal{
+                float *mask_buffer = static_cast<float *>(mask_data->at(i)->buffer());
+                py::list poly_batch_list;
+                for(unsigned j = prev_object_cnt; j < bbox_labels->at(i)->dims().at(0) + prev_object_cnt; j++)
+                {
```
Applicable to all such instances in the function

swetha097 commented on rocAL/rocAL/source/meta_data/bounding_box_graph.cpp:
```diff
@@ -26,6 +26,7 @@ void BoundingBoxGraph::process(pMetaDataBatch input_meta_data, pMetaDataBatch ou
     for (auto &meta_node : _meta_nodes)
     {
         meta_node->update_parameters(input_meta_data, output_meta_data);
+        input_meta_data = output_meta_data->clone();
```
Have we tested this with other trainings too ?

SundarRajan28 replied:
Ran classification and mask trainings with this change, @fiona-gladwin can u comment on the SSD training perf with this change here?

fiona-gladwin replied:
Ran Retinanet training:
CPU backend : Not much affected with this change
GPU backend : In progress

swetha097 commented on rocAL/rocAL/include/meta_data/meta_data.h:
```diff
@@ -170,14 +173,15 @@ class BoundingBox : public Label
         _mask_cords = std::move(mask_cords);
         _polygon_count = std::move(polygon_count);
         _vertices_count = std::move(vertices_count);
+        _info.img_id = img_id;
```
Any particular reason why std::move was not used for img_id ?

swetha097 commented on rocAL/rocAL/source/meta_data/coco_meta_data_reader.cpp:
```diff
@@ -72,7 +82,7 @@ void COCOMetaDataReader::lookup(const std::vector<std::string> &image_names)
 }
 
-void COCOMetaDataReader::add(std::string image_name, BoundingBoxCords bb_coords, Labels bb_labels, ImgSize image_size, MaskCords mask_cords, std::vector<int> polygon_count, std::vector<std::vector<int>> vertices_count)
+void COCOMetaDataReader::add(std::string image_name, BoundingBoxCords bb_coords, Labels bb_labels, ImgSize image_size, MaskCords mask_cords, std::vector<int> polygon_count, std::vector<std::vector<int>> vertices_count, int image_id)
```
Can this image_id be -ve ?
If not we can used unsigned int for image_id

fiona-gladwin replied:
Image ID is always negative. When we read from json it is an int value not unsigned hence maintaining it as int.
In open source also it was int only

swetha097 commented on rocAL/rocAL/source/meta_data/coco_meta_data_reader.cpp:
```diff
@@ -183,6 +193,7 @@ void COCOMetaDataReader::read_all(const std::string &path)
     {
         if (0 == std::strcmp(key, "images"))
         {
+            int image_id;
```
Why is this declared inside the while loop ?
Move it outside

SundarRajan28 replied:
The scope of this variable should be limited to when the images info are read from the json so I added the declaration here

swetha097 reviewed (commented):
Added the second round of review, Please address them

SundarRajan28 committed:
Resolving review comments

fiona-gladwin committed:
Add ROI tensor changes

Remove tensor variable from node.h file
Add a vx_tensor for ROI in every tensor object
Add API to create the ROI tensor from handle

fiona-gladwin committed:
Add support to create vx_scalar for layout and ROI type in each node

SundarRajan28 committed:
Reformatting jupyter notebooks

SundarRajan28 committed:
Merge remote-tracking branch 'origin/tensor_pybind_PR' into lk/rocalTensor_pybind

SundarRajan28 committed:
Merge remote-tracking branch 'origin/lk/rocalTensor_pybind' into PR_metanode_reader_changes

fiona-gladwin committed:
Add changes to inherit crop augmentations from CropNode

Add API to create crop tensor in crop node
Remove destrutors in other crop nodes

fiona-gladwin committed:
Minor change

SundarRajan28 committed:
Merge remote-tracking branch 'origin/PR_metanode_reader_changes' into PR_python_examples_notebooks

SundarRajan28 committed:
Merge remote-tracking branch 'origin/PR_metanode_reader_changes' into PR_tensor_mask

SundarRajan28 committed:
Resolving review comments

SundarRajan28 committed:
Adding C++ unit tests

swetha097 committed:
Code Clean up - Jupyter NB

swetha097 committed:
Add a genric path to the tf jupyter example

swetha097 committed:
Fix the dttpes and layouts in api_pipeline & video_pipeline

swetha097 committed:
Fix dtype and layout for cmn in coco pipeline.py

SundarRajan28 committed:
Merge remote-tracking branch 'origin/PR_tensor_node_augmentations' into lk/rocalTensor_pybind

SundarRajan28 committed:
Merge remote-tracking branch 'origin/lk/rocalTensor_pybind' into PR_metanode_reader_changes

SundarRajan28 committed:
Using metadata clone() only for multiple metanodes

SundarRajan28 committed:
Merge remote-tracking branch 'origin/PR_metanode_reader_changes' into PR_tensor_mask

SundarRajan28 committed:
Uncommenting RMN metanode

SundarRajan28 committed:
Uncommenting RMN metanode

SundarRajan28 committed:
Adding augmentation API parameters similar to DALI

SundarRajan28 committed:
Adding max ROI support for copyOutTensor kernels

SundarRajan28 committed:
Changing renamed params in examples

SundarRajan28 committed:
Optimizing aspect ratio grouping implementation

SundarRajan28 committed:
Resolving review comments

swetha097 committed:
Update image_augmentation.cpp

SundarRajan28 committed:
Reverting AR grouping to original implementation

SundarRajan28 committed:
Fixing bug in rocal HIP kernels

SundarRajan28 committed:
Adding optimized AR grouping implementation

SundarRajan28 committed:
Merge remote-tracking branch 'lk_dev/lk/rocalTensor' into lk/rocalTensor_pybind

SundarRajan28 committed:
Merge remote-tracking branch 'origin/lk/rocalTensor_pybind' into PR_metanode_reader_changes

SundarRajan28 committed:
Merge remote-tracking branch 'origin/PR_metanode_reader_changes' into PR_python_examples_notebooks

SundarRajan28 committed:
Fixing build issues with video reader

SundarRajan28 committed:
Fixing build issues with video reader

swetha097 committed:
Fix PR comments

SundarRajan28 committed:
Merge remote-tracking branch 'origin/tensor_pybind_PR' into lk/rocalTensor_pybind

fiona-gladwin committed:
Merge branch 'swetha/cpp_example_files' of https://github.com/fiona-gladwin/MIVisionX into swetha/cpp_example_files

SundarRajan28 committed:
Merge remote-tracking branch 'origin/lk/rocalTensor_pybind' into PR_metanode_reader_changes

SundarRajan28 committed:
Fixing bug with metadata graph process for non geometric nodes

SundarRajan28 committed:
Removing warning in metagraph process

SundarRajan28 committed:
Merge remote-tracking branch 'origin/PR_metanode_reader_changes' into PR_python_examples_notebooks

SundarRajan28 committed:
Merge remote-tracking branch 'lk_dev/lk/rocalTensor' into PR_metanode_reader_changes

SundarRajan28 committed:
Merge remote-tracking branch 'origin/PR_metanode_reader_changes' into PR_python_examples_notebooks

SundarRajan28 committed:
Adding updated examples and notebooks

SundarRajan28 committed:
Renaming memory types in python API

SundarRajan28 committed:
Resolving review comments

SundarRajan28 committed:
Merge remote-tracking branch 'origin/tensor_pybind_PR' into PR_metanode_reader_changes

fiona-gladwin committed:
Merge remote-tracking branch 'origin/PR_metanode_reader_changes' into swetha/cpp_example_files

SundarRajan28 committed:
Merge branch 'tensor_pybind_PR' into PR_python_examples_notebooks

fiona-gladwin committed:
Minor changes in Unit Tests

SundarRajan28 committed:
Minor changes in python examples

SundarRajan28 committed:
Removing unused variables from Node class

SundarRajan28 committed:
Fixing bug in video loader shutdown

SundarRajan28 committed:
Adding rocal release to video pipeline example

SundarRajan28 committed:
Merge branch 'PR_metanode_reader_changes' into PR_python_examples_notebooks

SundarRajan28 committed:
Merge remote-tracking branch 'lk_dev/lk/rocalTensor' into PR_metanode_reader_changes

SundarRajan28 committed:
Merge branch 'PR_metanode_reader_changes' into PR_python_examples_notebooks

swetha097 committed:
Merge remote-tracking branch 'lk_fork/lk/rocalTensor' into swetha/cpp_example_files

fiona-gladwin committed:
Merge remote-tracking branch 'lk_fork/lk/rocalTensor' into swetha/cpp_example_files

fiona-gladwin committed:
Merge branch 'swetha/cpp_example_files' of https://github.com/fiona-gladwin/MIVisionX into swetha/cpp_example_files

SundarRajan28 committed:
Resolving review comments

SundarRajan28 committed:
Merge remote-tracking branch 'origin/swetha/cpp_example_files' into cleanup_PR

SundarRajan28 committed:
Adding cpp and python formatting changes

SundarRajan28 committed:
Formatting cpp unit tests and examples

SundarRajan28 committed:
Adding changes to image comparison script for cpp unittests

fiona-gladwin committed:
Add Doxygen descriptor to python files

SundarRajan28 committed:
Modifying python files for doxygen comments and formatting

fiona-gladwin committed:
Minor changes

fiona-gladwin committed:
Merge branch 'cleanup_PR' of https://github.com/fiona-gladwin/MIVisionX into cleanup_PR

swetha097 committed:
Update image_augmentation.cpp

swetha097 committed:
Update open_cv_decoder.cpp

SundarRajan28 committed:
Changing default param value for SSD random crop

SundarRajan28 committed:
Merge branch 'PR_metanode_reader_changes' into PR_python_examples_notebooks

SundarRajan28 committed:
Merge remote-tracking branch 'lk_dev/lk/rocalTensor' into PR_python_examples_notebooks

SundarRajan28 committed:
Merge remote-tracking branch 'lk_dev/lk/rocalTensor' into cleanup_PR

SundarRajan28 committed:
Fixing build issue

SundarRajan28 committed:
Fixing build issue

SundarRajan28 committed:
Adding back label reading to tf reader

SundarRajan28 committed:
Merge remote-tracking branch 'lk_dev/lk/rocalTensor' into PR_tensor_mask

SundarRajan28 committed:
Resolving review comments

SundarRajan28 committed:
Merge branch 'PR_python_examples_notebooks' into PR_tensor_mask

fiona-gladwin committed:
Merge branch 'lk/rocalTensor' of https://github.com/LakshmiKumar23/MIVisionX into cleanup_PR

fiona-gladwin committed:
Address review comments

fiona-gladwin committed:
Address review comments

fiona-gladwin committed:
Change Aligment of comments
output file name change CropCenterCMN
keep rgb as 1  in image_augmentation_app

fiona-gladwin committed:
Minor changes

fiona-gladwin committed:
Merge branch 'cleanup_PR' of https://github.com/fiona-gladwin/MIVisionX into cleanup_PR

SundarRajan28 committed:
Merge branch 'lk/rocalTensor' into PR_tensor_mask

SundarRajan28 committed:
Merge branch 'cleanup_PR' into PR_tensor_mask

SundarRajan28 committed:
Adding formatting changes to rocal_pybind
